typescript/sdk/src/deploy/middleware/TokenBridgeRouterDeployer.ts

```diff
@@ -0,0 +1,224 @@
+import { ethers } from 'ethers';
+
+import {
+  CircleBridgeAdapter__factory,
+  TokenBridgeRouter__factory,
+} from '@hyperlane-xyz/core';
+import { objMap, promiseObjAll } from '@hyperlane-xyz/sdk/src/utils/objects';
+import { utils } from '@hyperlane-xyz/utils';
+
+import { chainMetadata } from '../../consts/chainMetadata';
+import { HyperlaneCore } from '../../core/HyperlaneCore';
+import { MultiProvider } from '../../providers/MultiProvider';
+import {
+  TokenBridgeContracts,
+  TokenBridgeFactories,
+  tokenBridgeFactories,
+} from '../../tokenBridge';
+import { ChainMap, ChainName } from '../../types';
+import { HyperlaneRouterDeployer } from '../router/HyperlaneRouterDeployer';
+import { RouterConfig } from '../router/types';
+
+export enum BridgeAdapterType {
+  Mock = 'Mock',
+  Circle = 'Circle',
+}
+
+export interface CircleBridgeAdapterConfig {
+  type: BridgeAdapterType.Circle;
+  circleBridgeAddress: string;
+  messageTransmitterAddress: string;
+  usdcAddress: string;
+  circleDomainMapping: {
+    hyperlaneDomain: number;
+    circleDomain: number;
+  }[];
+}
+
+interface MockTokenBridgeAdapterConfig {
+  type: BridgeAdapterType.Mock;
+  mockTokenAddress: string;
+}
+
+export type BridgeAdapterConfig =
+  | CircleBridgeAdapterConfig
+  | MockTokenBridgeAdapterConfig;
+
+export type TokenBridgeConfig = RouterConfig & {
+  bridgeAdapterConfigs: BridgeAdapterConfig[];
+};
+
+export class TokenBridgeDeployer<
+  Chain extends ChainName,
+> extends HyperlaneRouterDeployer<
+  Chain,
+  TokenBridgeConfig,
+  TokenBridgeContracts,
+  TokenBridgeFactories
+> {
+  constructor(
+    multiProvider: MultiProvider<Chain>,
+    configMap: ChainMap<Chain, TokenBridgeConfig>,
+    protected core: HyperlaneCore<Chain>,
+    protected create2salt = 'tokenbridgedeployersalt',
+  ) {
+    super(multiProvider, configMap, tokenBridgeFactories, {});
+  }
+
+  async enrollRemoteRouters(
+    contractsMap: ChainMap<Chain, TokenBridgeContracts>,
+  ): Promise<void> {
+    // Enroll the TokenBridgeRouter with each other
+    await super.enrollRemoteRouters(contractsMap);
+
+    // Enroll the circle adapters with each other
+    const deployedChains = Object.keys(contractsMap);
```

Comment:
isn't this basically just `RouterDeployer.enrollRemoteRouters` for the `CircleBridgeAdapter`?

Reply:
Yes it is, but I didn't want to extract the logic out to be independent of the key access. But now that you said it, i guess i can just pass a modified "contractsMap"